src/machines/activity_state_machine.rs

```diff
@@ -497,12 +497,17 @@ pub(super) struct StartedActivityCancelEventRecorded {}
 impl StartedActivityCancelEventRecorded {
     pub(super) fn on_activity_task_completed(
         self,
+        dat: SharedState,
         attrs: ActivityTaskCompletedEventAttributes,
     ) -> ActivityMachineTransition<Completed> {
-        ActivityMachineTransition::ok(
-            vec![ActivityMachineCommand::Complete(attrs.result)],
-            Completed::default(),
-        )
+        // We don want to tell lang about completions if we already pre-emptively canceled the
```

Comment:
I would say that in abandon and trycancel, workflow is not waiting for the activity to complete so we don't need to send activation. Also I believe abandon can not get here so it is just try cancel.

Reply:
Yep, that's exactly what this is doing. The empty list means no activation gets sent.